backend/analysis/views.py

```diff
@@ -0,0 +1,137 @@
+                description="All worked hours for each student for a certain period",
+                response=inline_serializer(
+                    name="WorkedHoursAnalysisResponse",
+                    fields={
+                        "student_id": serializers.IntegerField(),
+                        "worked_minutes": serializers.IntegerField(),
+                        "student_on_tour_ids": serializers.ListField(child=serializers.IntegerField()),
+                    },
+                ),
+                examples=[
+                    OpenApiExample(
+                        "Successful Response 1",
+                        value={"student_id": 6, "worked_minutes": 112, "student_on_tour_ids": [1, 6, 9, 56, 57]},
+                    ),
+                    OpenApiExample(
+                        "Successful Response 2",
+                        value={"student_id": 7, "worked_minutes": 70, "student_on_tour_ids": [2, 26]},
+                    ),
+                ],
+            ),
+        },
+    )
+    def get(self, request):
+        """
+        Get all worked hours for each student for a certain period
+        """
+        student_on_tour_instances = StudentOnTour.objects.all()
+        filters = {
+            "start_date": get_filter_object("date__gte", required=True),
+            "end_date": get_filter_object("date__lte", required=True),
+            "region_id": get_filter_object("tour__region__id"),
+        }
+
+        try:
+            student_on_tour_instances = filter_instances(request, student_on_tour_instances, filters)
+        except BadRequest as e:
+            return bad_request_custom_error_message(str(e))
+
+        serializer = self.serializer_class()
+        serialized_data = serializer.to_representation(student_on_tour_instances)
+        return Response(serialized_data, status=status.HTTP_200_OK)
+
+
+class StudentOnTourAnalysis(APIView):
+    permission_classes = [IsAuthenticated, IsAdmin | IsSuperStudent]
+    serializer_class = StudentOnTourAnalysisSerializer
+
+    @extend_schema(
+        description="Get a detailed view on a student on tour's timings",
+        responses={
+            200: OpenApiResponse(
+                description="A list of buildings and their timings on this student on tour",
+                response=inline_serializer(
+                    name="DetailedStudentOnTourTimings",
+                    fields={
+                        "building_id": serializers.IntegerField(),
+                        "expected_duration_in_seconds": serializers.IntegerField(),
+                        "arrival_time": serializers.DateTimeField(),
+                        "departure_time": serializers.DateTimeField(),
+                        "duration_in_seconds": serializers.IntegerField(),
+                    },
+                ),
+                examples=[
+                    OpenApiExample(
+                        "Successful Response 1",
+                        value={
```

Comment:
same here, serializers return a list, but the example shows a single object

Reply:
The list is correct. But I couldn't do that in the docs. Not sure why.